homeassistant/components/tuya/climate.py

```diff
@@ -187,9 +252,61 @@ def supported_features(self):
     @property
     def min_temp(self):
         """Return the minimum temperature."""
-        return self._tuya.min_temp()
+        min_temp = (
+            self._min_temp if self._min_temp is not None else self._tuya.min_temp()
+        )
+        if min_temp is not None:
+            return min_temp
+        return super().min_temp
 
     @property
     def max_temp(self):
         """Return the maximum temperature."""
-        return self._tuya.max_temp()
+        max_temp = (
+            self._max_temp if self._max_temp is not None else self._tuya.max_temp()
+        )
+        if max_temp is not None:
+            return max_temp
+        return super().max_temp
+
+    def _get_ext_temperature(self):
+        if not self._temp_entity:
+            return None
+
+        def _log_error(error_msg):
+            if not self._temp_entity_error:
+                _LOGGER.warning(
+                    "Error on Tuya external temperature sensor %s: %s",
+                    self._temp_entity,
+                    error_msg,
+                )
+                self._temp_entity_error = True
+
+        try:
+            attr_list = self._temp_entity.split(".", 2)
+            if len(attr_list) < 2:
+                _log_error("entity name is invalid")
+                return None
```

Comment:
Validate this at configure time instead so we don't let the invalid data into the config.

Reply:
Should I check also that sensor exist or only the syntax?

Reply:
Since its in the config flow we expect HA to already be started so it seems fine check for existence as well.  Ideally you could borrow the code from `homekit`s config flow and allow them to choose the entity in the config flow so they don't mistype it.

We should keep the existence check here as well in case it disappears though

Reply:
I did it, anyway this is limiting a little bit the options compared with the original implementation where was possible to also select an entity attribute instead of the attribute state. In any case I think that this will avoid a lot of typing errors and so it is acceptable.

Reply:
Home Assistant is moving away from having sensors as additional attributes.  If the integration hasn't been updated yet, it can be wrapped with a template sensor.